crates/curp/tests/it/common/curp_group.rs

```diff
@@ -332,6 +333,49 @@ impl CurpGroup {
             .all(|node| node.task_manager.is_finished())
     }
 
+    pub async fn wait_for_node_shutdown(&self, node_id: u64, duration: Duration) {
+        let node = self
+            .nodes
+            .get(&node_id)
+            .expect("{node_id} should exist in nodes");
+        let res = std::iter::once(node);
+        timeout(duration, Self::wait_for_targets_shutdown(res))
+            .await
+            .expect("wait for group to shutdown timeout");
+        assert!(
+            node.task_manager.is_finished(),
+            "The target node({node_id}) is not finished yet"
+        );
+    }
+
+    pub async fn wait_for_group_shutdown(&self, duration: Duration) {
+        timeout(
+            duration,
+            Self::wait_for_targets_shutdown(self.nodes.values()),
+        )
+        .await
+        .expect("wait for group to shutdown timeout");
+        assert!(self.is_finished(), "The group is not finished yet");
+    }
+
+    async fn wait_for_targets_shutdown(targets: impl Iterator<Item = &CurpNode>) {
+        let final_tasks: [TaskName; 3] = [
+            TaskName::WatchTask,
+            TaskName::ConfChange,
+            TaskName::LogPersist,
```

Comment:
These are bottom tasks, which are waited by all the other tasks. I recommend to put them into a group so that we won't miss any of them even we refactor the task dependency topo later.

Reply:
Done